superset/views/base_api.py

```diff
@@ -84,6 +88,72 @@ def wraps(self: "BaseSupersetModelRestApi", *args: Any, **kwargs: Any) -> Respon
     return functools.update_wrapper(wraps, f)
 
 
+def get_level(status: int) -> ErrorLevel:
+    if status < 400:
+        return ErrorLevel.INFO
+    if status < 500:
+        return ErrorLevel.WARNING
+    return ErrorLevel.ERROR
+
+
+def handle_exception(f: Callable[..., FlaskResponse]) -> Callable[..., FlaskResponse]:
+    """
+    A decorator that formats exceptions.
+
+    SIP-40 (https://github.com/apache/superset/issues/9194) introduced
+    a standard error payload for all API errors returned by Superset.
+    This decorator formats exceptions to conform to it.
+    """
+
+    def wraps(self: Any, *args: Any, **kwargs: Any) -> FlaskResponse:
+        try:
+            return f(self, *args, **kwargs)
+        except SupersetErrorException as ex:
+            logger.warning(ex)
+            return json_errors_response(errors=[ex.error], status=ex.status)
+        except SupersetErrorsException as ex:
+            logger.warning(ex)
+            return json_errors_response(errors=ex.errors, status=ex.status)
+        except CommandInvalidError as ex:
+            logger.warning(ex)
+            return json_errors_response(
+                errors=[
+                    SupersetError(
```

Comment:
Are we always going to throw HTTP 500 on these? would it make sense that the exception itself could identify the HTTP status code?

Reply:
I don't think that command exceptions should have HTTP status codes, they should care only about business logic and be decoupled from the API.

Ideally the API layer should do the mapping between command exceptions to `SupersetError(s)Exception`. The reason why I added `CommandInvalidError` and `CommandException` is so we can gradually implement that mapping, while in the meantime any non-covered command exceptions return a 500.